manifests/charts/istio-control/istio-discovery/files/gen-istio.yaml

```diff
@@ -889,6 +896,39 @@ spec:
       - name: config-volume
         configMap:
           name: istio
+      affinity:
+        nodeAffinity:
+            requiredDuringSchedulingIgnoredDuringExecution:
```

Comment:
why do we need required *and* preferred?

Reply:
Both the "hard" and "soft" requirements are put here to ensure istiod will be scheduled to node with listed arch, this also align with other components(istio-ingress/egress gateways), see: https://github.com/istio/istio/blob/master/manifests/charts/gateways/istio-ingress/templates/_affinity.tpl#L11-L43

Reply:
Doesn't the hard requirement take precedence? I am not sure how the soft one can ever be used

Reply:
We can specify different `weight` for each soft requirement, then the istiod pod will be scheduled to node with highest **weight** that matched the hard requirement.